### DISCUSSION TITLE
Unique constraint failed on the constraint: `_id_` in MongoDriver

### DISCUSSION THREAD

**Discussion by Draccano** [prisma/prisma#20001] [Q&A]
### Bug description

Hi, I have recently faced an issue I could not resolve in the last few hours. I have created Mongodb schema for the User entity. When I call create method on User, It throws an error
```
api:dev: The entity already exists PrismaClientKnownRequestError: 
api:dev: Invalid `.create()` invocation in
api:dev: c:\Bohemi Interactive\feedbacktracker\feedback-tracker\apps\api\src\modules\users\infra\http\routes\index.ts:23:8
api:dev:
api:dev:   20   idHex: id.toHexString(),
api:dev:   21 });
api:dev:   22 const resp = await prismaClient.user
api:dev: → 23   .create(
api:dev: Unique constraint failed on the constraint: `_id_`
api:dev:     at Object.handleRequestError (c:\Bohemi Interactive\feedbacktracker\feedback-tracker\node_modules\.pnpm\@prisma+client@4.16.1_prisma@4.15.0\node_modules\@prisma\client\runtime\library.js:174:7325)
api:dev:     at Object.handleAndLogRequestError (c:\Bohemi Interactive\feedbacktracker\feedback-tracker\node_modules\.pnpm\@prisma+client@4.16.1_prisma@4.15.0\node_modules\@prisma\client\runtime\library.js:174:6754)
api:dev:     at Object.request (c:\Bohemi Interactive\feedbacktracker\feedback-tracker\node_modules\.pnpm\@prisma+client@4.16.1_prisma@4.15.0\node_modules\@prisma\client\runtime\library.js:174:6344)
api:dev:     at createTest (c:\Bohemi Interactive\feedbacktracker\feedback-tracker\apps\api\src\modules\users\infra\http\routes\index.ts:22:18)
api:dev:     at mainReqHandler (c:\Bohemi Interactive\feedbacktracker\feedback-tracker\node_modules\.pnpm\@ts-rest+express@3.26.3_dk52cgqmpvibef4w6qveug34r4\node_modules\@ts-rest\express\index.js:82:28) {
api:dev:   code: 'P2002',
api:dev:   clientVersion: '4.16.1',
api:dev:   meta: { target: '_id_' }
api:dev: }
```
Even if the validator throws an error, the document is finally stored in DB. Weird things on that are, that if I switch from manual id binding to auto() or if I change the name of the model to anything else than `User` with same properties, it works. I also tried reset collections or clearing caches. Tried on atlas mongo instance and even on docker one nothing helped.

### How to reproduce

1. Create prisma schema as above
2. Call create for user
3. Check whether it stored

### Expected behavior

user is created without validation _id_ issue.

### Prisma information

```prisma
datasource db {
    provider = "mongodb"
    url      = env("MONGO_URI")
}

generator client {
    provider = "prisma-client-js"
}

model User {
    id              String    @id @map("_id") @db.ObjectId
    email           String    @unique
    isEmailVerified Boolean   @default(false)
    isDeleted       Boolean   @default(false)
    username        String?   @unique
    password        String?
    role            RoleTypes

    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt
    @@map("users")
}

enum RoleTypes {
    REGULAR
    MODERATOR
    DEVELOPER
    ADMIN
}

```

```js
// Prisma client singleton
import { PrismaClient as PrismaClientLib } from "@prisma/client";

class PrismaClient {
  static instance = null;

  static getInstance(): PrismaClientLib {
    if (!PrismaClient.instance) {
      PrismaClient.instance = new PrismaClientLib();
    }
    return PrismaClient.instance;
  }
}

export default PrismaClient;
```

```js
 const id = new ObjectId();
 const prismaClient = PrismaClient.getInstance();
 //...
 const resp = await prismaClient.user
      .create({
        data: {
          id: id.toString(),
          email,
          password,
          username,
          role,
        },
      }).catch(err => console.log(err);
```

### Environment & setup

- OS: Windows
- Database: MongoDB 
- Node.js version: v18.12.0

### Prisma Version

```
4.16.1
```

**Comment by Draccano** [comment]
I did a little more playground and found out it works if I init Prisma client directly from lib instead of my own singleton, I don't know why 🤔. My motivation was to initialize PrismaClient once for the whole app.

```js
import { PrismaClient } from "@prisma/client";

const prismaClient = new PrismaClient();
 //...
 const id = new ObjectId();
 const resp = await prismaClient.user
      .create({
        data: {
          id: id.toString(),
          email,
          password,
          username,
          role,
        },
      }).catch(err => console.log(err);

```

**Comment by jkomyno** [comment]
Hi @Draccano, as it seems you have already found an alternative solution, I will convert this issue into a discussion, so that my colleagues can perhaps help you identifying why one snippet works and the other doesn't.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
